### ISSUE TITLE
FlareChain's response_chain not picking up specified LLM model

### ISSUE THREAD

**Issue by devinbost** [langchain-ai/langchain#10493]
### System Info

Using LangChain 0.0.276
Python 3.11.4 

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [X] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [X] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Construct a FlareChain instance like this and run it:

```
    myllm = ChatOpenAI(temperature=0, model_name="gpt-3.5-turbo-16k")

    flare = FlareChain.from_llm(
        llm=myllm,
        retriever=vectorstore.as_retriever(),
        max_generation_len=164,
        min_prob=0.3,
    )
    result = flare.run(querytext)
```

When I inspect during debugging, the specified LLM model was set on `flare.question_generator_chain.llm.model_name` but NOT `flare.response_chain.llm.model_name`,
which is still the default value. 

### Expected behavior

I'm expecting `flare.response_chain.llm.model_name` to return `gpt-3.5-turbo-16k`, not `text-davinci-003`

**Comment by devinbost** [comment]
We also need a way to support the AzureOpenAI class. If I do this:

 ```
       if llm._llm_type == "azure":
            response_llm = AzureOpenAI(
                max_tokens=max_generation_len,
                model_name=llm.model_name,
                model_kwargs={"logprobs": 1}, temperature=0
            )
        else:
            response_llm = OpenAI(
                max_tokens=max_generation_len,
                model_name=llm.model_name,
                model_kwargs={"logprobs": 1}, temperature=0
            )
```
it uses the protected `_llm_type` field, plus it's clunky when we need to add more types, and I'd need to pass in additional fields since this implementation is still incomplete. 

One option would be for me to update OpenAI to support a prototype pattern so that I could construct the `response_chain` llm from the provided `question_generator_chain` llm. 
It looks like the only difference otherwise is that the new llm instance needs `model_kwargs={"logprobs": 1}`
Is there anything else I'm missing? 

If this is the right approach, I can get a new PR cut for this quickly.

**Comment by devinbost** [comment]
When I try setting:
```
           response_llm = OpenAI(
               max_tokens=max_generation_len,
               model_name=llm.model_name,
               model_kwargs={"logprobs": 1}, temperature=0
           )
```
the response_llm is an OpenAIChat instance, not an OpenAI instance, so it doesn't have the logprobs required for FLARE.

**Comment by devinbost** [comment]
I'm seeing this in the OpenAI constructor:

```
    def __new__(cls, **data: Any) -> Union[OpenAIChat, BaseOpenAI]:  # type: ignore
        """Initialize the OpenAI object."""
        model_name = data.get("model_name", "")
        if model_name.startswith("gpt-3.5-turbo") or model_name.startswith("gpt-4"):
            warnings.warn(
                "You are trying to use a chat model. This way of initializing it is "
                "no longer supported. Instead, please use: "
                "`from langchain.chat_models import ChatOpenAI`"
            )
            return OpenAIChat(**data)
        return super().__new__(cls)
```
Since it looks like `gpt-3.5-turbo-16k` is a chat model, we would need to get logprobs from the newer chat models that support more tokens... But, it looks like that is blocked upstream: https://community.openai.com/t/logprobs-in-chatcompletion/329471

**Comment by tarek-kerbedj** [comment]
@devinbost  did you manage to find a workaround for that problem ?

**Comment by devinbost** [comment]
No, it looks like this is blocked by the upstream LLM not supporting logprobs yet.

**Comment by vaidehiu** [comment]
can I add open llm hugging face model as part of this because its asking OPEN AI key I want to use open LLM model instead

**Comment by akashAD98** [comment]
please update package by this & use llm ,it will work 
```
!pip install -U langchain-openai
```
```
from langchain_openai import ChatOpenAI
llm = ChatOpenAI(temperature=0, model_name="gpt-3.5-turbo-16k")
```
